src/Core/Services/IFeatureService.cs

```diff
@@ -1,6 +1,39 @@
-﻿namespace Bit.Core.Services;
+﻿using Bit.Core.Context;
+
+namespace Bit.Core.Services;
 
 public interface IFeatureService
 {
+    /// <summary>
+    /// Checks whether online access to feature status is available.
+    /// </summary>
+    /// <returns>True if the service is online, otherwise false.</returns>
     bool IsOnline();
+
+    /// <summary>
+    /// Checks whether a given feature is enabled.
+    /// </summary>
+    /// <param name="key">The key of the feature to check.</param>
+    /// <param name="currentContext">A context providing information that can be used to evaluate whether a feature should be on or off.</param>
+    /// <param name="defaultValue">The default value for the feature.</param>
+    /// <returns>True if the feature is enabled, otherwise false.</returns>
+    bool IsEnabled(string key, ICurrentContext currentContext, bool defaultValue = false);
```

Comment:
Can we use an enum as the key to prevent someone from accidentally misspelling something or creating multiple flags for 1 thing?